File: Contiguous Array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int>m;
        m[0]=0;
        int sum=0;
        int out=0;
        for(int i=0;i<n;i++)
        {
            int x=nums[i];
            sum+=x==0?-1:1;
            if(m.find(sum)!=m.end())
            {
                out=max(out,i-m[sum]+1);
            }
            else
            {
                m[sum]=i+1;
            }
        }
        return out;
    }
};

File: Golang Solutions/Contiguous Array.go
func findMaxLength(nums []int) int {
    n:=len(nums)
    m:=make(map[int]int)
    m[0]=0;
    sum:=0;
    out:=0;
    for i:=0;i<n;i++{
        if nums[i]==0{
            sum+=-1
        }else{
            sum+=1
        }
        a,ok:=m[sum]
        if ok{
            if out<i-a+1{
                out=i-a+1
            }
        }else{
            m[sum]=i+1;
        }
    }
    return out;
}